JWAnimatedImage/JWAnimatedImageView.swift

```diff
@@ -29,14 +29,15 @@ public extension UIImageView{
         self.init()
         AddGifImage(gifImage,manager: manager,loopTime: loopTime);
     }
-    
+
     public func AddGifImage(gifImage:UIImage,manager:JWAnimationManager){
         // -1 means always run
         AddGifImage(gifImage,manager: manager,loopTime: -1);
     }
-    
+
     public func AddGifImage(gifImage:UIImage,manager:JWAnimationManager,loopTime:Int){
         self.loopTime = loopTime
+        self.animationManager?.DeleteImageView(self)
```

Comment:
Hi,@kirualex
Add this line may cause some logic problem.
I'm prefer improving function function `isDiscarded` or `isDisplayedInScreen` instead of modifying current state.
Because when GIF view is out of screen,it will turn state display into state suspend.
Then,when it waked up.All it need to do,is modify bool factor `displaying` from `false` to `true`.
But if you add this code,it will be deleted and re-init.
So,I don't think this is a good choose.
You can focus on  improving function function `isDiscarded` or `isDisplayedInScreen`.
If you still have problems with it,just let me know.But I may need a demo again to fix this issue because it does not appear in any my project.